code-challenges.js

```diff
@@ -7,24 +7,68 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+// Function that takes in a string
+const codedMessage = (str) => {
+  // Split the string into an array
+  let strArray = str.split("")
+  // Map through array
+  let codedArray = strArray.map(value=> {
+    // Convert the appropriate vowels to numbers
+    if (value === "a") return "4"
+    else if (value === "e") return "3"
+    else if (value === "i") return "1"
+    else if (value === "o") return "0"
+    // Return all other values as itself
+    else return value
+  })
+  // Join back into string
+  return codedArray.join("")
+}
 
-
+console.log(codedMessage(secretCodeWord1))
+console.log(codedMessage(secretCodeWord2))
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
+// Function that takes in array
+const hasLetterA = (arr) => {
+  // Filter through the array
+  return arr.filter(word => {
+    // Convert word to lowercase and into an array
+    let wordArray = word.toLowerCase().split("")
+    // Check if the array includes "a", if so, return
+    return wordArray.includes("a")
+  })
+}
```

Comment:
you can also just use .includes on the whole string! no need to split :) here's what I mean

```javascript
const hasLetterA = (arr) => {
  return arr.filter(word => {
    return word.toLowerCase().includes("a")
  })
}
console.log(hasLetterA(arrayOfWords))
```